tests/test_order_SolidLiquid.py

```diff
@@ -0,0 +1,79 @@
+import numpy.testing as npt
+import freud
+import unittest
+import util
+
+# Validated against manual calculation and pyboo
+PERFECT_FCC_Q6 = 0.57452416
+
+
+class TestSolidLiquid(unittest.TestCase):
+    def test_shape(self):
+        N = 1000
+        L = 10
+
+        box, positions = util.make_box_and_random_points(L, N)
+
+        comp = freud.order.SolidLiquid(6, Q_threshold=.7, S_threshold=6)
+        comp.compute(box, positions, query_args=dict(r_max=2.0))
+
+        npt.assert_equal(comp.cluster_idx.shape, (N,))
+
+    def test_identical_environments(self):
+        box, positions = util.make_fcc(4, 4, 4)
+
+        comp_default = freud.order.SolidLiquid(
+            6, Q_threshold=.7, S_threshold=6)
+        comp_no_norm = freud.order.SolidLiquid(
+            6, Q_threshold=.3, S_threshold=6, normalize_Q=False)
+
+        for comp in (comp_default, comp_no_norm):
+            for query_args in (dict(r_max=2.0), dict(num_neighbors=12)):
+                comp.compute(box, positions, query_args=query_args)
+                self.assertEqual(comp.largest_cluster_size, len(positions))
+                self.assertEqual(len(comp.cluster_sizes), 1)
+                self.assertEqual(comp.cluster_sizes[0], len(positions))
+                npt.assert_array_equal(comp.num_connections, 12)
+
+    def test_attribute_access(self):
+        box, positions = util.make_fcc(4, 4, 4)
+        sph_l = 6
+        Q_threshold = 0.7
+        S_threshold = 4
+        normalize_Q = False
+
+        comp = freud.order.SolidLiquid(
+            sph_l, Q_threshold=Q_threshold, S_threshold=S_threshold,
+            normalize_Q=normalize_Q)
+
+        self.assertEqual(comp.l, sph_l)
+        npt.assert_allclose(comp.Q_threshold, Q_threshold)
+        npt.assert_allclose(comp.S_threshold, S_threshold)
+        self.assertEqual(comp.normalize_Q, normalize_Q)
+
+        with self.assertRaises(AttributeError):
+            comp.largest_cluster_size
+        with self.assertRaises(AttributeError):
+            comp.cluster_sizes
+        with self.assertRaises(AttributeError):
+            comp.cluster_idx
+        with self.assertRaises(AttributeError):
+            comp.num_connections
+        with self.assertRaises(AttributeError):
+            comp.plot()
+
+        comp.compute(box, positions, query_args=dict(r_max=2.0))
+
+        comp.largest_cluster_size
+        comp.cluster_sizes
+        comp.cluster_idx
+        comp.num_connections
+        comp._repr_png_()
+
+    def test_repr(self):
+        comp = freud.order.SolidLiquid(6, Q_threshold=.7, S_threshold=6)
+        self.assertEqual(str(comp), str(eval(repr(comp))))
+
+
```

Comment:
While I know this isn't really the point of this code (since it's supposed to be insensitive to what type of structure you have), there are known characteristic values for FCC that were in the paper, right? Could we test against that? Also this would be a good system to add a test against a known simulation output by just storing a final frame in a numpy array or so as a testing dataset.

Reply:
I am not exposing the Ql(i, j) to Python so I can't test this. At one point I was, but I don't think it helps the user API because it's a per-bond quantity. Then we would have to expose a NeighborList to make that quantity usable, etc.

Reply:
That's fair. This raises a different point though; is there no concept of per-particle value? The determination of whether a bond is solid checks whether each bond is solid like using the S threshold. Is the per-particle determination a useful quantity? It seems like it would be, although it's effectively given by belonging to a cluster unless there is a singleton cluster (which wouldn't get caught).

Reply:
The per-particle values are Qlmi (which is just from Steinhardt) and the number of solid-like bonds (which is used to filter the bonds to only include (i, j) iff both i and j have S or more solid-like bonds).

Reply:
I test the number of solid-like bonds in the `num_connections` property, and the `Qlmi` values aren't exposed. We're covered well enough here.